esp-hal/src/gpio.rs

```diff
@@ -1863,9 +1865,21 @@ pub struct IO {
 }
 
 impl IO {
-    pub fn new(mut gpio: GPIO, io_mux: IO_MUX) -> Self {
-        gpio.bind_gpio_interrupt(gpio_interrupt_handler);
+    /// Initialize the I/O driver.
+    pub fn new(gpio: GPIO, io_mux: IO_MUX) -> Self {
+        Self::new_with_priority(gpio, io_mux, crate::interrupt::Priority::min())
+    }
 
+    /// Initialize the I/O driver with a interrupt priority.
+    ///
+    /// This decides the priority for the interrupt when only using async.
+    pub fn new_with_priority(
```

Comment:
not super stoked about this API, but I'm not unhappy about it either, suggestions welcome!